examples/python/interceptors/headers/header_manipulator_client_interceptor.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Interceptor that adds headers to outgoing requests."""
+
+import collections
+
+import grpc
+from generic_client_interceptor import GenericClientInterceptor as _GenericClientInterceptor
+
+class _ClientCallDetails(
+        collections.namedtuple('_ClientCallDetails',
+                               ('method', 'timeout', 'metadata',
+                                'credentials')), grpc.ClientCallDetails):
+    pass
+
+
+class HeaderManipulatorClientInterceptor(_GenericClientInterceptor):
```

Comment:
We've managed to avoid implementation inheritance in gRPC Python so far; why should this class be something that is a GenericClientInterceptor rather than something that has a GenericClientInterceptor?

Reply:
Redesigned. Thanks.